src/compiler/checker.ts

```diff
@@ -34287,10 +34291,69 @@ namespace ts {
                     return isLiteralImportTypeNode(node) ? getSymbolAtLocation(node.argument.literal) : undefined;
 
                 case SyntaxKind.ExportKeyword:
-                    return isExportAssignment(node.parent) ? Debug.assertDefined(node.parent.symbol) : undefined;
+                    if (isExportAssignment(node.parent)) {
+                        return Debug.assertDefined(node.parent.symbol);
+                    }
+                    break;
+            }
+        }
 
-                default:
-                    return undefined;
+        /**
+         * Gets the symbol related to the provided location, if it that location is a keyword.
+         * These additional keywords are normally only used to resolve references but would
+         * not be used for document highlights, quickinfo, etc.
+         */
+        function getSymbolAtKeyword(node: Node): Symbol | undefined {
+            if (node.flags & NodeFlags.InWithStatement) {
+                // We cannot answer semantic questions within a with block, do not proceed any further
+                return undefined;
+            }
+
+            const { parent } = node;
+
+            // If the node is a modifier of its parent, get the symbol for the parent.
+            if (isModifier(node) && contains(parent.modifiers, node)) {
+                return getSymbolOfNode(parent);
+            }
+
+            switch (node.kind) {
+                case SyntaxKind.InterfaceKeyword:
+                case SyntaxKind.EnumKeyword:

```

Comment:
I think you also need to handle const enum ? Don't remember if const is modifier for that enum and so already handled